File: frontend/zeal-edu/src/app/core/models/user-account.ts
import { CandidateInfo } from './candidate-info';
import { FacultyInfo } from './faculty-info';
import { Gender } from './gender';
import { StaffInfo } from './staff-info';
import { UserRole } from './user-role';

export interface UserAccount {
  id: string;
  username: string;
  fullName: string;
  email: string;
  phone: string;
  dob: string;
  gender: Gender;
  role: UserRole;
  isActive: boolean;
  lastLogin: string | null;
  staff?: StaffInfo;
  faculty?: FacultyInfo;
  candidate?: CandidateInfo;
}

File: backend/src/ZealEducation.Domain/Entities/UserAccount.cs
using ZealEducation.Domain.Common;
using ZealEducation.Domain.Enums;

namespace ZealEducation.Domain.Entities;

public class UserAccount : BaseAuditableEntity
{
    public string Username { get; set; } = default!;
    public string PasswordHash { get; set; } = default!;
    public string FullName { get; set; } = default!;
    public string Email { get; set; } = default!;
    public string Phone { get; set; } = default!;
    public DateOnly Dob { get; set; }
    public Gender Gender { get; set; }
    public UserRole Role { get; set; }
    public bool IsActive { get; set; } = true;
    public bool MustChangePassword { get; set; }
    public int FailedLoginCount { get; set; }
    public DateTime? LastLogin { get; set; }
}
